.buildkite/pipeline.yml

```diff
@@ -5,6 +5,17 @@ use_plugins: &use_plugins
           bucket: "a8c-repo-mirrors"
           repo: "wordpress-mobile/wordpress-ios/"
 
+xcode_version: &xcode_version
+  IMAGE_ID: xcode-12.5.1
+
+env: &env
+  # If you'll need more than one key-value, update to:
+  #
+  # env:
+  #   <<: *xcode_version
+  #   other_key: other_value
+  env: *xcode_version
+
```

Comment:
1. In your comment to suggest how to handle more than one value, you forgot to set the anchor: `env: &env`
2. Overall I'm not sure it's worth extracting the `xcode-version:` (and its `IMAGE_ID` key) entry separately as the only place it's used is in the env…
So what about merging them already, allowing you to get rid of the comment altogether?

```suggestion
x-common-env: &env
  env:
    IMAGE_ID: xcode-12.5.1
```

PS: I use `x-common-env` as the key just to emphasis that the name of this key:

1. Is purely arbitrary, and doesn't have to be `env:` here, because what the `&env` YAML anchor captures is what's after `&env` (so the key is not captured by the anchor, which is why you have to specify the `env:` key after/inside `&env` if you want it to be outputed when you use `*env` to "paste" it later in the YAML)
2. Is not a key that has a specific meaning in the BuildKite specification – hence starting it with `x-` in an attempt to convey the fact that it's a custom key from us, not a BuildKite key, similar to the old (and now deprecated, but 🤷 ) convention that HTTP used for experimental/non-standard headers)

---

Another approach would be to not include the `env:` key inside the `&env` anchor, and instead repeat it at every call site explicitly. That kinda reduces the goal of the anchor DRY-ing stuff and avoiding repeating that `env:` key every time, but I think that it would actually look nicer in this context, since this anchor definition here would then look like this:

```suggestion
env: &env
  IMAGE_ID: xcode-12.5.1
```
And at call site like [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/16929/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53L16-R27) instead of using the YAML `<<:` merge operator, it would instead look like this, which is fine imho:
```yml
command: ".buildkite/commands/build-for-testing.sh"
env: &env
<<: *use_plugins
```

In any case, both solutions work so I'll let you decide which one you prefer (it doesn't really matter to me I'm fine with both 😉), as long as it gets simplified as I don't think we need the `xcode_version` to be extracted as opposed to being directly defined inline in the `env` dictionary we DRY'd here.